File: 1812/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(squareIsWhite("a1"))
	fmt.Println(squareIsWhite("h3"))
	fmt.Println(squareIsWhite("c7"))
}

func squareIsWhite(coordinates string) bool {
	c1 := 'a' - coordinates[0] + 1
	c2 := coordinates[1]
	return (c1 + c2) % 2 != 0 
}

File: 1812/main.rs
struct Solution {}

impl Solution{
    pub fn square_is_white(coordinates: String) -> bool {
        

        let c: Vec<char> = coordinates.chars().collect();
        let c1 = 'a' as i32 - c[0] as i32 + 1;
        let c2 = c[1] as i32;
        (c1 + c2) % 2 != 0 
    }
}

fn main() {
    println!("{}", Solution::square_is_white("h3".to_string()));
}